src/Controller/Component/AuthenticationComponent.php

```diff
@@ -89,6 +109,47 @@ public function beforeFilter()
         ], $this->getController());
     }
 
+    /**
+     * Start up event handler
+     *
+     * If `requireIdentity` is true, the action will be compared to the
+     * allowed actions.
+     *
+     * @return void
+     */
+    public function startup()
+    {
+        if (!$this->getConfig('requireIdentity')) {
+            return;
+        }
+        $request = $this->getController()->request;
+        $action = $request->getParam('action');
+        if (in_array($action, $this->unauthenticatedActions)) {
+            return;
+        }
+
+        $identity = $request->getAttribute('identity');
+        if (!$identity) {
+            throw new UnauthorizedException([]);
+        }
+    }
+
+    /**
+     * Set the list of actions that don't require an authentication identity to be present.
+     *
+     * Actions not in this list will require an identity to be present. Any
+     * valid identity will pass this constraint.
+     *
+     * @param array $actions The action list.
+     * @return $this
+     */
+    public function allowUnauthenticated(array $actions)
```

Comment:
Should this have a merge option to allow adding to existing list? Unless there's a method to retrieve existing list so that user can merge himself and reset.

Reply:
For e.g. I would like to allow 'view' always through AppController and then additional action inside a specific controller.

Reply:
Would it then merge by default? Calling the param $overwrite (default false) like in other places in core?

Reply:
Why a getter? Wouldnt this be

    allowUnauthenticated(array $actions, $merge = false)

?

Reply:
I think I'd prefer a seperate method for this. Or we could just utilise instance config for this `$this->Authentication->setConfig('allowUnauthenticated', ['index']);`

Reply:
Handling/Merging on this level is not a good idea IMO. Better to have either dedicated methods or just the bool 2nd param.

Reply:
Right. `mergeUnauthenticated(['index'])` is cleaner in terms of what it does than `allowUnauthenticated(['index'], true)`

Reply:
Well, you could always do `allowUnauthenticated(['index'], ['merge' => true])` :)

Reply:
Merging to existing list will be the most common usage. So it should be possible to do that with minimal code :slightly_smiling_face:.

Reply:
So mergeUnauthenticated() is the method name to move forward here?

Reply:
Added.